pkg/snapshot/publisher.go

```diff
@@ -69,86 +78,107 @@ func (p *Publisher) PublishHeader(header *types.Header) (int64, error) {
-		stateKey = node.Key.Hex()
-	}
-	tx, err := p.db.Beginx()
-	if err != nil {
-		return 0, err
+	if !bytes.Equal(node.key.Bytes(), nullHash.Bytes()) {
+		stateKey = node.key.Hex()
 	}
+
 	defer func() {
-		if p := recover(); p != nil {
+		if rec := recover(); rec != nil {
 			shared.Rollback(tx)
-			panic(p)
-		} else if err != nil {
-			shared.Rollback(tx)
-		} else {
-			err = tx.Commit()
+			panic(rec)
 		}
 	}()
-	stateCIDStr, err := shared.PublishRaw(tx, ipld.MEthStateTrie, multihash.KECCAK_256, node.Value)
+
+	stateCIDStr, mhKey, err := shared.PublishRaw(tx, ipld.MEthStateTrie, multihash.KECCAK_256, node.value)
 	if err != nil {
 		return 0, err
 	}
-	mhKey, _ := shared.MultihashKeyFromCIDString(stateCIDStr)
+
 	err = tx.QueryRowx(`INSERT INTO eth.state_cids (header_id, state_leaf_key, cid, state_path, node_type, diff, mh_key) VALUES ($1, $2, $3, $4, $5, $6, $7)
- 									ON CONFLICT (header_id, state_path) DO UPDATE SET state_path = state_cids.state_path
+ 									ON CONFLICT (header_id, state_path) DO UPDATE SET (state_leaf_key, cid, node_type, diff, mh_key) = ($2, $3, $5, $6, $7)
  									RETURNING id`,
-		headerID, stateKey, stateCIDStr, node.Path, node.NodeType, false, mhKey).Scan(&stateID)
+		headerID, stateKey, stateCIDStr, node.path, node.nodeType, false, mhKey).Scan(&stateID)
+
+	p.currBatchSize += 2
 	return stateID, err
 }
 
 // PublishStorageNode writes the storage node to the ipfs backing pg datastore and adds secondary indexes in the storage_cids table
-func (p *Publisher) PublishStorageNode(node Node, stateID int64) error {
+func (p *Publisher) PublishStorageNode(node *node, stateID int64, tx *sqlx.Tx) error {
 	var storageKey string
-	if !bytes.Equal(node.Key.Bytes(), nullHash.Bytes()) {
-		storageKey = node.Key.Hex()
-	}
-	tx, err := p.db.Beginx()
-	if err != nil {
-		return err
+	if !bytes.Equal(node.key.Bytes(), nullHash.Bytes()) {
+		storageKey = node.key.Hex()
 	}
+
 	defer func() {
-		if p := recover(); p != nil {
+		if rec := recover(); rec != nil {
 			shared.Rollback(tx)
-			panic(p)
-		} else if err != nil {
-			shared.Rollback(tx)
-		} else {
-			err = tx.Commit()
+			panic(rec)
 		}
 	}()
-	storageCIDStr, err := shared.PublishRaw(tx, ipld.MEthStorageTrie, multihash.KECCAK_256, node.Value)
+
+	storageCIDStr, mhKey, err := shared.PublishRaw(tx, ipld.MEthStorageTrie, multihash.KECCAK_256, node.value)
 	if err != nil {
 		return err
 	}
-	mhKey, _ := shared.MultihashKeyFromCIDString(storageCIDStr)
-	_, err = tx.Exec(`INSERT INTO eth.storage_cids (state_id, storage_leaf_key, cid, storage_path, node_type, diff, mh_key) VALUES ($1, $2, $3, $4, $5, $6, $7) 
- 							  ON CONFLICT (state_id, storage_path) DO NOTHING`,
-		stateID, storageKey, storageCIDStr, node.Path, node.NodeType, false, mhKey)
-	return err
+
+	_, err = tx.Exec(`INSERT INTO eth.storage_cids (state_id, storage_leaf_key, cid, storage_path, node_type, diff, mh_key) VALUES ($1, $2, $3, $4, $5, $6, $7)
```

Comment:
We're updating to the new version of go-ethereum on the `postgres_refactor` branch, but these statements are formatted for the old schemas.